feat: add --check CLI option

smably opened the pull request:
My use case: I want to run `prettier --check` on CI, but in order to speed up my CI tasks, I'd rather not check every file in my codebase on every commit. Ideally I'd like to be able to run `prettier --check`, but just on the files changed in the current branch.

This is really similar to what `pretty-quick --bail` does, but I don't actually want to format the files, I just want pretty-quick to tell me if they need formatting.

This PR adds a `--check` CLI flag that runs [`prettier.check`](https://prettier.io/docs/en/api.html#prettierchecksource-options) instead of `prettier.format` and exits with a non-zero status code if any files fail the check.

I also renamed `onWriteFile` to `onProcessFile` and `formatFiles.js` to `processFiles.js` to reflect that they now handle either formatting _or_ checking operations. This change is in a separate commit to keep the diff for the [main code change](https://github.com/azz/pretty-quick/commit/d4e9f762cb7daf2f73aff3dae78f5469b849c939) a bit cleaner.

azz commented on src/processFiles.js:
```diff
@@ -0,0 +1,35 @@
+export default (
+  directory,
+  files,
+  { check, config, onProcessFile, onExamineFile } = {}
```
Names `onProcessFile`, `onExamineFile` are a little too similar for my liking.

How about:
- `onProcessFile` - file is being looked at
- `onWriteFile` - file was just written
- `onCheckFile` - file was just checked - always call with `(file, isFormatted)`

smably replied:
This makes sense! Fixed.

azz commented on src/processFiles.js:
```diff
@@ -0,0 +1,35 @@
+    );
+    const input = readFileSync(file, 'utf8');
+
+    if (check && !prettier.check(input, options)) {
```
Maybe something like this?

```js
    if (check) {
      const isFormatted = prettier.check(input, options);
      onCheckFile && onCheckFile(relative, isFormatted);
      return;
    }
```

smably replied:
Done!

smably commented:
I tested this a bit more and realized my initial implementation was buggy. 😳

Fixed this up a bunch, renamed a few things, also made it so only the failed filenames are printed (same as how filenames are only printed when files are written).

azz commented on src/index.js:
```diff
@@ -14,12 +14,14 @@ export default (
+    onProcessFile,
+    onCheckFile,
     onWriteFile,
-    onExamineFile,
```
Sorry to be a pain, but we can prevent this from being an API breaking change by keeping it called `onExamineFile` instead of `onProcessFile`. Not sure if anyone uses this module's API, but doesn't hurt to not break it.

smably replied:
No problem! So you're thinking in the public API (i.e., just `index.js`), this would remain `onExamineFile`, but internally it would be called `onProcessFile` (per https://github.com/azz/pretty-quick/pull/64#discussion_r284496821)? Or would you prefer to call these `onExamineFile`, `onWriteFile`, and `onCheckFile` across the board?

azz replied:
The latter

smably replied:
Done!

azz commented:
Thanks!
